packages/frontend/src/components/wallet-migration/WalletMigration.jsx

```diff
@@ -95,7 +109,7 @@ const WalletMigration = ({ open, onClose }) => {
                     <MigrationSecret
                         showMigrationPrompt={showMigrationPrompt}
                         showMigrateAccount={showMigrateAccount}
-                        secretKey={keyToString(initialState.migrationKey)}
+                        secretKey={state.wallet.id === 'sender' ? pinCode : keyToString(initialState.migrationKey)}
```

Comment:
[Same as above](https://github.com/near/near-wallet/pull/2957/files#r1007528714) in here

instead of inline, case statement will be easier for other wallets/teams to easily update their custom `secretKey` logic here.